### PR TITLE
Updates to use recent changes made to ethrpc.

### PR DESCRIPTION
**Pull request by MicahZoltu**
`setLatestBlock` has been replaced by `ethrpc`'s `ensureLatestBlock`.

Details about connectivity are no longer provided by ethrpc, so `connection` is now just a true or false (it was not used anywhere previously for more than a falsy check).

`ethrpc` now has a `connect` method rather than a bunch of properties that need to be set.  This is now called instead of setting up those properties (which no longer exist).

### CODE REVIEW

**Comment by tinybike** [comment]
I npm linked the new ethrpc with ethereumjs locally and ran `npm test`, which gave the following errors:
```
  1) setCoinbase error if blockchain coinbase is 0x, coinbase and from unchanged sync:

      AssertionError: expected [Function: TypeError] to equal [Function: Error]
      + expected - actual

      -[Function]
      +{
      +  "prepareStackTrace": [undefined]
      +  "stackTraceLimit": Infinity
      +}
      
      at Function.assert.strictEqual (node_modules/chai/lib/chai/interface/assert.js:178:32)
      at Object.assertions (test/connect.js:970:14)
      at Context.<anonymous> (test/connect.js:842:13)

  2) setCoinbase error if blockchain coinbase is 0x, coinbase and from unchanged async:
     TypeError: Cannot read property 'constructor' of null
      at Object.assertions (test/connect.js:970:29)
      at test/connect.js:853:13
      at index.js:143:57
      at Object.ethcon.rpc.coinbase (test/connect.js:849:11)
      at Object.setCoinbase (index.js:140:16)
      at Context.<anonymous> (test/connect.js:852:16)

  3) setCoinbase error if blockchain coinbase is null, coinbase and from unchanged sync:

      AssertionError: expected [Function: TypeError] to equal [Function: Error]
      + expected - actual

      -[Function]
      +{
      +  "prepareStackTrace": [undefined]
      +  "stackTraceLimit": Infinity
      +}
      
      at Function.assert.strictEqual (node_modules/chai/lib/chai/interface/assert.js:178:32)
      at Object.assertions (test/connect.js:994:14)
      at Context.<anonymous> (test/connect.js:842:13)

  4) setCoinbase error if blockchain coinbase is null, coinbase and from unchanged async:
     TypeError: Cannot read property 'constructor' of null
      at Object.assertions (test/connect.js:994:29)
      at test/connect.js:853:13
      at index.js:142:31
      at Object.ethcon.rpc.coinbase (test/connect.js:849:11)
      at Object.setCoinbase (index.js:140:16)
      at Context.<anonymous> (test/connect.js:852:16)

  5) sync connect sync connection sequence with api:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

**Comment by MicahZoltu** [comment]
Fixed.

That was my bad.  One of the last changes I made was to make having a coinbase no longer a requirement.  I didn't run the tests after that change, so I didn't notice that there were some tests that were asserting an error would occur.

**Comment by tinybike** [comment]
Getting there!  There's still one error being thrown:
```
  1) sync connect sync connection sequence with api:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```
Possibly something isn't being properly stubbed?

**Comment by MicahZoltu** [comment]
Ah, that test takes ~2.5 seconds for me, running with no mocha timeout.  I can add a mocha-opts file and increase the timeout, but the fundamental problem is that running a synchronous test against a remote host is slow.  :/  This is why there is only one sync test.  :'(  The async tests run in milliseconds due to being able to use the stub server.

I have gone ahead and added a `mocha.opts` file with a 3 second timeout.  Let me know if you would rather take a different approach to solving this particular problem.

**Comment by tinybike** [comment]
Still failing for me... :(
```
  1) sync connect sync connection sequence with api:
     Error: Timeout of 3000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

**Comment by tinybike** [comment]
Ok, just needed to be increased a little more I guess:
```
  sync connect
    ✓ sync connection sequence with api (3882ms)
```

File: index.js [hunk 1]
```diff
@@ -210,18 +184,16 @@ module.exports = {
       this.setGasPrice.bind(this)
     ], function (err) {
       if (err) return self.retryConnect(err, options, callback);
-      self.state.connection = {
-        http: self.rpc.nodes.local || self.rpc.nodes.hosted,
-        ws: self.rpc.wsUrl,
-        ipc: self.rpc.ipcpath
-      };
+      self.state.connection = true;
```
Comment (tinybike) [inline index.js]: My preference is to revert this change, if possible -- I found it quite useful having a `connection` object come back from `connect()` with the HTTP, WS, and/or IPC endpoints I'm connected to.

Comment (MicahZoltu) [inline index.js]: What is the use case for the results?  When I made this change, I looked through the code in Augur.js and Augur UI and didn't see anything using the results other than for a truthy check.

If the only use case is for debugging, then as I mentioned elsewhere, my recommendation is to not include code in production that is only for debugging, especially when that information is readily available in the browser.  That being said, if the desired use case is in fact debug type logging then perhaps the best way to handle this is to have Transporter `console.log` directly when it is finished connecting with the endpoints/transports that it is connected to?  This could be put behind the `rpc.debug.broadcast` flag so it is off by default but can be turned on?

If there really is a need to have connection information propagated all the way up the chain that will take a bit of work as right now only Transporter knows what ethrpc is actually connected to (and all logic for switching/choosing is in Transporter).

**Comment by tinybike** [comment]
Ah, I didn't notice you mentioned this in the original PR.  My mistake.

> That being said, if the desired use case is in fact debug type logging then perhaps the best way to handle this is to have Transporter console.log directly when it is finished connecting with the endpoints/transports that it is connected to?

Yes, it's basically just for debugging.  This sounds like a good solution.  It's nice having the endpoints print out to the console as soon as you open the UI -- makes it very obvious if you've accidentally connected to the wrong endpoint.

**Comment by MicahZoltu** [comment]
I have made it so `if (rpc.debug.broadcast)` then it will log connection details to the console once the transports are selected (letting you know what all you ended up being connected to).
